cmd/otel-allocator/main.go

```diff
@@ -148,10 +152,12 @@ func main() {
 	runGroup.Add(
 		func() error {
 			// Initial loading of the config file's scrape config
-			err = targetDiscoverer.ApplyConfig(allocatorWatcher.EventSourceConfigMap, cfg.Config)
-			if err != nil {
-				setupLog.Error(err, "Unable to apply initial configuration")
-				return err
+			if cfg.Config != nil {
+				err = targetDiscoverer.ApplyConfig(allocatorWatcher.EventSourceConfigMap, cfg.Config)
```

Comment:
I wonder whether it would make more sense to check for `cfg.Config == nil` inside of `ApplyConfig` and if so, just do early return

Reply:
I can't reverse it if it stays where it is in `main`, as `cfg.Config` == `nil` by itself does not imply an error. Or do you mean doing this inside `ApplyConfig`?

Reply:
wouldn't `cfg.Config == nil` imply an error is i guess my conjecture here. i.e. if we don't have a config to apply at all we should fail fast.

Reply:
@matej-g makes sense, you've convinced me.

@jaronoff97 Having the raw Prometheus config be nil and Prometheus CR handling enabled looks like a valid configuration to me. That's why I added the validate method and check for this specific state.

Reply:
I moved the check to ApplyConfig and added a test for this.